func/func_server.cc

```diff
@@ -0,0 +1,54 @@
+#include "func_server.h"
+
+FuncServiceImpl::FuncServiceImpl() {
+  warble_ = new WarbleService();
+}
+
+Status FuncServiceImpl::hook(ServerContext* context, const HookRequest* request, HookReply* reply) {
+  int event_type = request->event_type();
+  std::string event_function = request->event_function();
+  std::pair<int, std::string> pair(event_type, event_function);
+  map_.insert(pair);
+  return Status::OK;
+}
+
+Status FuncServiceImpl::unhook(ServerContext* context, const UnhookRequest* request, UnhookReply* reply) {
+  int event_type = request->event_type();
+  auto it = map_.find(event_type);
+  if (it == map_.end()) {
+    return Status(StatusCode::NOT_FOUND, "Event not found");
+  } else {
+    return Status::OK;
+  }
+}
+
+Status FuncServiceImpl::event(ServerContext* context, EventRequest* request, EventReply* reply) {
+  int event_type = request->event_type();
+  auto it = map_.find(event_type):
+  if (it == map_.end()) {
+    return StatusCode::NOT_FOUND, "Event not found");
+  } else {
```

Comment:
Are you missing an right bracket? I will suggest compile your code first for each pull request.

Reply:
At the same time, in this case, I would suggest removing the else, since this function has already returned on error condition.